test/suite/wtbackup.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-2021 MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+import os, glob
+import wttest, wiredtiger
+from suite_subprocess import suite_subprocess
+from helper import compare_files
+
+# Shared base class used by backup tests.
+class backup_base(wttest.WiredTigerTestCase, suite_subprocess):
+    cursor_config = None        # a config string for cursors
+    mult = 0                    # counter to have variance in data
+    nops = 100                  # number of operations added to uri
+
+    # counter to used to produce unique backup ids the backup id, and is
+    # generally used only for using add_data() first time.
+    counter = 0
+    # To determine whether to increase/decrease counter, which determines
+    initial_backup = True
+    # Used for populate function
+    rows = 100
+    populate_big = None
+
+    #
+    # Add data to the given uri.
+    # Allows the option for doing a session checkpoint after adding data.
+    #
+    def add_data(self, uri, key, val, do_checkpoint=False):
+        assert(self.nops != 0)
+        c = self.session.open_cursor(uri, None, self.cursor_config)
+        for i in range(0, self.nops):
+            num = i + (self.mult * self.nops)
+            k = key + str(num)
+            v = val + str(num)
+            c[k] = v
+        c.close()
+        if do_checkpoint:
+            self.session.checkpoint()
+        # Increase the counter so that later backups have unique ids.
+        if not self.initial_backup:
+            self.counter += 1
+        # Increase the multiplier so that later calls insert unique items.
+        self.mult += 1
+    
```

Comment:
There's a minor thing that I think you'll see if you do this in your tree:
```
$ git diff develop test/suite/wtbackup.py
```
There are a few places where extra spaces at the end of lines are shown highlighted in that file.  Also you can detect by:
```
$ grep -n ' $' test/suite/wtbackup.py 
```
I wouldn't care too much, except that `git diff` seems to care, so the diffs don't look very good.

Reply:
It appears that this is also the case in `test_backup17.py`. Whitespace in `test_backup17.py` and `wtbackup.py` are causing the PR testing to fail. So this does need to be fixed.

Reply:
Sorry forgot to run s_all! i think it fixes s_all